google/cloud/bigtable/testing/table_integration_test.cc

```diff
@@ -105,7 +110,43 @@ void TableIntegrationTest::SetUp() {
       TableTestEnvironment::project_id(), TableTestEnvironment::instance_id(),
       ClientOptions());
 
-  ASSERT_STATUS_OK(table_admin_->DropAllRows(TableTestEnvironment::table_id()));
+  // We cannot use `DropAllRows()` to cleanup the table because the integration
+  // tests against production sometimes use all the "'DropRowRangeGroup' and
+  // limit 'USER-100s'" quota. Instead we delete the rows by using a
+  BulkMutation bulk;
+  auto table = GetTable();
+  // Bigtable does not support more than 100,000 mutations in a BulkMutation,
+  // if we had that many rows then just fallback on DropAllRows(). Most tests
+  // only have a small number of rows, so this is a good strategy to save
+  // DropAllRows() quota, and should be fast in most cases.
+  std::size_t const maximum_mutations = 100000;
+
+  for (auto const& row :
+       table.ReadRows(RowRange::InfiniteRange(), Filter::PassAllFilter())) {
+    if (not row) {
+      break;
+    }
+    bulk.emplace_back(
+        SingleRowMutation(row->row_key(), bigtable::DeleteFromRow()));
+    if (bulk.size() > maximum_mutations) {
+      break;
+    }
+  }
+
+  // The version of the emulator distributed with the Cloud SDK does not handle
+  // deleted rows correctly:
+  //   https://github.com/googleapis/google-cloud-go/issues/1240
+  // this has been fixed at HEAD, but the changes have not made it to a version
+  // we can use yet. So just use DropAllRows() in that case.
+  if (bulk.size() > maximum_mutations || UsingCloudBigtableEmulator()) {
+    ASSERT_STATUS_OK(
+        table_admin_->DropAllRows(TableTestEnvironment::table_id()));
```

Comment:
Is it worth clearing `bulk` first?

Reply:
I do not believe so, is is a local variable, the destructor will do the same thing.

Reply:
The idea is to not hold on to the 100,001 mutations during the `DropAllRows()`, but if the latter only has a trivial memory footprint then I guess it doesn't matter.